### ISSUE TITLE
Support Authentication

### ISSUE THREAD

**Issue by wzalazar** [graphitejs/server#157]
https://www.apollographql.com/docs/apollo-server/features/authentication

**Comment by rodrigooler** [comment]
@wzalazar in this feature will contain the context implementation also correct?

**Comment by wzalazar** [comment]
@rodrigooler yes, I think is just add the context option. If you want you can contribute to this feature...

The best way to start with that is create a functional test, for example:

https://github.com/graphitejs/server/blob/master/test/functional/mutation.js

where Graphite could receive the `context option`

```
const context = ({ req }) => {
   // get the user token from the headers
   const token = req.headers.authorization || '';
  
   // try to retrieve a user with the token
   const user = getUser(token);
  
   // add the user to the context
   return { user };
 }

const graphite = await Graphite({ models: [Developer], context })
```

btw, you can run the functional test with this command 

`npm run test:functional`

**Comment by wzalazar** [comment]
if you start with this issue and you have doubts or problems, ping me!

**Comment by rodrigooler** [comment]
@wzalazar this problem seems very simple. I am just trying to reconcile with my coding schedule and I will do the PR with the implementation and also discuss some possible points of improvement in the code.
